lib/cfn-nag/custom_rules/DMSEndpointPasswordRule.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require 'cfn-nag/util/enforce_reference_parameter'
+require 'cfn-nag/util/enforce_string_or_dynamic_reference'
+require_relative 'base'
+
+class DMSEndpointPasswordRule < BaseRule
+  def rule_text
+    'DMS Endpoint password must be Ref to NoEcho Parameter. ' \
```

Comment:
let's flip this rule_text around.  DMS endpoint must not be plaintext string of a Ref to a Noecho parm with default.  we're letting the dynamic references by or anything else we don't "understand"